### PR TITLE
new improved CallableMixin

### PR DESCRIPTION
**Pull request by RootShinobi**
# Description

new improved CallableMixin

## Type of change

Please delete options that are not relevant.

- [ ] Documentation (typos, code examples or any documentation update)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

all internal aiogram tests

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: CHANGES/1357.misc.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+new improved CallableMixin
```
Comment (JrooTJunior) [inline CHANGES/1357.misc.rst:1]: The description of the changes should answer the question "What was done?" and be understandable to users of the framework.
The given description doesn't answer this question and is not clear.

**Review by JrooTJunior — CHANGES_REQUESTED** [review]
I know you have a performance test and comparison of results, so please post that test and results as a comment to this pull request.

This can help advanced users understand what has actually been changed.

File: CHANGES/1357.misc.rst (hunk 1 above)
Comment (Olegt0rr) [inline CHANGES/1357.misc.rst:1]:
```suggestion
Processing of `CallableMixin` has been accelerated by caching references to nested objects.
```

**Comment by RootShinobi** [comment]
> I know you have a performance test and comparison of results, so please post that test and results as a comment to this pull request.
> 
> This can help advanced users understand what has actually been changed.

```python
from typing import Set, Dict, Any
import matplotlib.pyplot as plt
from aiogram.dispatcher.event.handler import CallableMixin, CallbackType
from dataclasses import field, dataclass
import inspect
from timeit import timeit


@dataclass
class CallableMixinEX:
    callback: CallbackType
    awaitable: bool = field(init=False)
    params: Set[str] = field(init=False)
    varkw: bool = field(init=False)

    def post_init(self) -> None:
        callback = inspect.unwrap(self.callback)
        self.awaitable = inspect.isawaitable(callback) or inspect.iscoroutinefunction(callback)
        spec = inspect.getfullargspec(callback)
        self.params = {*spec.args, *spec.kwonlyargs}
        self.varkw = spec.varkw is not None

    def _prepare_kwargs(self, kwargs: Dict[str, Any]) -> Dict[str, Any]:
        if self.varkw:
            return kwargs

        return {k: kwargs[k] for k in self.params if k in kwargs}


REPEAT = 1000000
MAX_FUNC_PARAMS = 10
MAX_MIDDLEWARE_DATA = MAX_FUNC_PARAMS + 5
KEYS = tuple(f"param_{i}" for i in range(MAX_MIDDLEWARE_DATA))
X = []
X_EX = []
Y = []

for FUNC_PARAMS_SIZE in range(1, MAX_FUNC_PARAMS):
    PARAMS = ", ".join(KEYS[:FUNC_PARAMS_SIZE])
    FUNC = eval(f"lambda {PARAMS}: 1")
    for MIDDLEWARE_SIZE in range(FUNC_PARAMS_SIZE, MAX_MIDDLEWARE_DATA):
        MIDDLEWARE_DATA = dict(zip(KEYS[:MIDDLEWARE_SIZE], range(MIDDLEWARE_SIZE)))
        X.append(timeit(
            stmt="call._prepare_kwargs(MIDDLEWARE_DATA)",
            setup="call = CallableMixin(FUNC)",
            globals={**globals(), **locals()},
            number=REPEAT
        ))

        X_EX.append(timeit(
            stmt="call._prepare_kwargs(MIDDLEWARE_DATA)",
            setup="call = CallableMixinEX(FUNC)",
            globals={**globals(), **locals()},
            number=REPEAT
        ))
        Y.append(f"{FUNC_PARAMS_SIZE}x{MIDDLEWARE_SIZE}")

plt.plot(Y, X)
plt.plot(Y, X_EX)
plt.xlabel('func param size x middleware data size')
plt.ylabel('time, [s]')
plt.grid()
plt.xticks(rotation=-90)
plt.legend(['old', 'new'])
plt.show()
```

photo

**Review by JrooTJunior — APPROVED** [review]

**Pull request merged**